### DISCUSSION TITLE
Request for a torch-version and tensor-based fanbeam CT SIRT

### DISCUSSION THREAD

**Discussion by GreameLee** [astra-toolbox/astra-toolbox#470] [Q&A]
I build my reconstructed methods based on diffusion and all the data and model need to be in cuda and can run in the format of tensor. But I find astra is based in numpy and can not connect with cuda based project.

**Comment by LiRoc007** [comment]
> I build my reconstructed methods based on diffusion and all the data and model need to be in cuda and can run in the format of tensor. But I find astra is based in numpy and can not connect with cuda based project.

You can try [Tomosipo](https://github.com/ahendriksen/tomosipo) to build your project. Hope it works.

**Reply by GreameLee** [reply]
Can this tool be applied for 2D such as parallel-beam or fan-beam reconstruction?

**Reply by LiRoc007** [reply]
Absolutely! Tomosipo is a pythonic wrapper for the ASTRA-toolbox, you can find more detials on [here]a(https://github.com/ahendriksen/tomosipo/tree/master/tomosipo/geometry).
As for the usage of 2D astra such as parallel-beam or fan-beam reconstruction you can see this [#468](https://github.com/astra-toolbox/astra-toolbox/discussions/468#discussioncomment-9332514)

**Comment by askorikov** [comment]
Indeed. See also https://github.com/astra-toolbox/astra-toolbox/discussions/416 and https://github.com/astra-toolbox/astra-toolbox/discussions/388.
